subprojects/maven/src/main/java/org/gradle/api/publish/maven/plugins/MavenPublishPlugin.java

```diff
@@ -126,30 +132,81 @@ private void realizePublishingTasksLater(final Project project, final Publishing
             createGenerateMetadataTask(tasks, publication, mavenPublications, buildDirectory);
             createGeneratePomTask(tasks, publication, buildDirectory, project);
             createLocalInstallTask(tasks, publishLocalLifecycleTask, publication);
-            createPublishTasksForEachMavenRepo(tasks, publishLifecycleTask, publication, repositories);
+            createPublishTasksForEachMavenRepo(project, tasks, publishLifecycleTask, publication, repositories);
         });
     }
 
     private String publishAllToSingleRepoTaskName(MavenArtifactRepository repository) {
         return "publishAllPublicationsTo" + capitalize(repository.getName()) + "Repository";
     }
 
-    private void createPublishTasksForEachMavenRepo(final TaskContainer tasks, final TaskProvider<Task> publishLifecycleTask, final MavenPublicationInternal publication, final NamedDomainObjectList<MavenArtifactRepository> repositories) {
+    private void createPublishTasksForEachMavenRepo(final Project project, final TaskContainer tasks, final TaskProvider<Task> publishLifecycleTask, final MavenPublicationInternal publication, final NamedDomainObjectList<MavenArtifactRepository> repositories) {
         final String publicationName = publication.getName();
         repositories.all(repository -> {
             final String repositoryName = repository.getName();
             final String publishTaskName = "publish" + capitalize(publicationName) + "PublicationTo" + capitalize(repositoryName) + "Repository";
-            tasks.register(publishTaskName, PublishToMavenRepository.class, publishTask -> {
+            TaskProvider<PublishToMavenRepository> publicationTask = tasks.register(publishTaskName, PublishToMavenRepository.class, publishTask -> {
                 publishTask.setPublication(publication);
                 publishTask.setRepository(repository);
                 publishTask.setGroup(PublishingPlugin.PUBLISH_TASK_GROUP);
                 publishTask.setDescription("Publishes Maven publication '" + publicationName + "' to Maven repository '" + repositoryName + "'.");
             });
+            project.getGradle().getTaskGraph().whenReady(graph -> {
+                if (graph.hasTask(getTaskPath(project, publicationTask))) {
```

Comment:
More for my education, and possibly a dumb question, but is there something stopping us from setting the `whenReady` action from within the configuration action above (and then we could use `hasTask(publishTask)`?

Reply:
Indeed, simplifies the code as well, thanks!